### PR TITLE
Throw specific exception when no data type specified for root group

### PR DESCRIPTION
**Pull request by rly**
This is helpful to address https://github.com/NeurodataWithoutBorders/pynwb/issues/1077 where an incompatible file is read.

### CODE REVIEW

**Comment by t-b** [comment]
@rly How about detecting NWBv1 files as well and give the user a good error message. If written and found something at https://github.com/AllenInstitute/ipfx/blob/664f9f3578fe9a34f27c80b742b2af91459521c7/ipfx/nwb_reader.py#L606.

**Comment by rly** [comment]
@t-b thanks for the code snippet. I had started to write something similar in pynwb which relies on catching the above exception, but maybe the NWB version should be checked before triggering the whole builder parsing logic. I'll put up a PR in pynwb soon for feedback.

**Review by oruebel — APPROVED** [review]

File: src/hdmf/backends/hdf5/h5tools.py [hunk 1]
```diff
@@ -1043,3 +1049,7 @@ def __rec_get_ref(self, l):
             else:
                 ret.append(elem)
         return ret
+
+
+class MissingRootException(ValueError):
```
Comment (ajtritt) [inline src/hdmf/backends/hdf5/h5tools.py:1054]: I don't think this should ever happen. Root always exists by definition.

**Review by ajtritt — CHANGES_REQUESTED** [review]
I'm not sure what the best solution is, but this seems like it could be a bandage on a larger problem. `root` could not have a `data_type` if you just added some objects (with `data_type`) to the root group, which seems like a perfectly okay thing to do.

**Comment by rly** [comment]
Rethinking this, I agree with @ajtritt that this is not the best approach to the problem.

**Pull request closed without merging**